src/app/productConfig/productConfigModal/productConfig.modal.component.js

```diff
@@ -235,6 +235,19 @@ class ProductConfigModalController {
       this.submittingGift = false;
     } );
   }
+
+  displayName() {
+    return this.productData.displayName === this.itemConfig['jcr-title'] ?
+      this.productData.displayName : this.itemConfig['jcr-title'];
```

Comment:
I believe you are correct, jcr-title will always be set and should be used for the displayName.